lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -228,15 +228,17 @@ def __init__(self, paths=None, scan_sys_paths=True):
             paths.extend(sys.path)
 
         good_paths = []
+        real_paths = set()
         # expand any placeholders in configured paths
         for p in paths:
-
-            # ensure we always have ansible_collections
-            if os.path.basename(p) == 'ansible_collections':
-                p = os.path.dirname(p)
-
-            if p not in good_paths and os.path.isdir(to_bytes(os.path.join(p, 'ansible_collections'), errors='surrogate_or_strict')):
-                good_paths.append(p)
+            if p not in good_paths:
+                # ensure we always have ansible_collections
+                if os.path.basename(p) == 'ansible_collections':
+                    p = os.path.dirname(p)
+                r = os.path.realpath(p)
```

Comment:
If `ansible_collections` is the symlink, that seems like it should be deduplicated as well.

```suggestion
                r = os.path.realpath(os.path.join(p, 'ansible_collections'))
```